pkg/mimir/grpc_health_check.go

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package mimir
+
+import (
+	"context"
+
+	"go.uber.org/atomic"
+	"google.golang.org/grpc/health/grpc_health_v1"
+)
+
+func newCompositeCheck(checks ...grpc_health_v1.HealthServer) grpc_health_v1.HealthServer {
```

Comment:
I don't like the verbosity here just to add an `atomic.Bool`. Thoughts about extending dskit's  `grpcutil.HealthServer` with this functionality or adding these implementations there?

Reply:
I don't have concern about this verbosity. I think what you did is nice and extensible. Because of this, I see it a good for dskit too. If you want to move there, I'm up to review it quickly (ping me).